pandas/io/excel/_odfreader.py

```diff
@@ -0,0 +1,165 @@
+import pandas
+from pandas.io.parsers import TextParser
+
+
+class _ODFReader(object):
+    """Read tables out of OpenDocument formatted files
+
+    Parameters
+    ----------
+    filepath_or_stream: string, path to be parsed or
+        an open readable stream.
+    """
+    def __init__(self, filepath_or_stream):
+        try:
+            from odf.opendocument import load as document_load
+            from odf.table import Table
+        except ImportError:
+            raise ImportError("Install odfpy for OpenDocument support")
+
+        self.filepath_or_stream = None
+        self.document = None
+        self.tables = None
+        self.filepath_or_stream = filepath_or_stream
+        self.document = document_load(filepath_or_stream)
```

Comment:
This is called `self.book` in the existing xlrd code - can we use the same naming convention here?

Reply:
That's a little harder case... in _xlrd._XlrdReader book looks to be a reference to the xlrd class that provides the interface to excel files.

odfpy provides more of an ElementTree or lxml level interface than anything useful about tables. self.tables is just a dictionary pointing to the root node of table XML nodes found in the OpenDocument file. I'm not sure its high level enough to be considered equivalent to a "book" object.

Reply:
OK - you'll have to excuse that I'm not terribly familiar with odfpy. This is a rather minor sticking point so not really a blocker for this PR but consistency in the codebase is very important; it *may* make sense subsequently to change the xlrd code to reflect this convention or come up with an abstraction for both that reads consistently